solver/src/solver/http_solver/settlement.rs

```diff
@@ -128,36 +128,11 @@ fn match_prepared_and_settled_amms(
             let prepared = prepared_orders
                 .remove(&index)
                 .ok_or_else(|| anyhow!("invalid amm {}", index))?;
-
-            // TODO - Having ConstantProductOrder with tuples and WeightedProductOrder with HashMap is making this unnecessarily complicated.
-            let tokens = prepared.tokens.get();
-            let updates = (
-                *settled
-                    .updates
-                    .get(&tokens.0)
-                    .expect("received update for unknown token"),
-                *settled
-                    .updates
-                    .get(&tokens.1)
-                    .expect("received update for unknown token"),
-            );
-            let (input, output) = if updates.0.is_positive() && updates.1.is_negative() {
-                (
-                    (tokens.0, i128_abs_to_u256(updates.0)),
-                    (tokens.1, i128_abs_to_u256(updates.1)),
-                )
-            } else if updates.1.is_positive() && updates.0.is_negative() {
-                (
-                    (tokens.1, i128_abs_to_u256(updates.1)),
-                    (tokens.0, i128_abs_to_u256(updates.0)),
-                )
-            } else {
-                return Err(anyhow!("invalid uniswap update {:?}", settled));
-            };
             Ok(ExecutedAmm {
                 order: prepared,
-                input,
-                output,
+                // This seems backwards, but its how we passed the test.
+                input: (settled.buy_token, settled.exec_buy_amount),
+                output: (settled.sell_token, settled.exec_sell_amount),
```

Comment:
I would have expected input to be the sell related stuff here and output to be buy-related stuff. However, this is what passes the unit test.

Reply:
This is modelled the point of view of an amm, where input=buy and output=sell.